rclpy/src/rclpy/_rclpy.c

```diff
@@ -1161,6 +1161,27 @@ rclpy_get_subscriptions_info_by_topic(PyObject * Py_UNUSED(self), PyObject * arg
   return _get_info_by_topic(args, "subscriptions", rcl_get_subscriptions_info_by_topic);
 }
 
+static PyObject *
+rclpy_get_resolved_subscription_name(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pysubscription;
+  if (!PyArg_ParseTuple(args, "O", &pysubscription)) {
+    return NULL;
+  }
+
+  rclpy_subscription_t * sub =
+    rclpy_handle_get_pointer_from_capsule(pysubscription, "rclpy_subscription_t");
+  if (NULL == sub) {
+    return NULL;
+  }
+
+  const char * subscription_name = rcl_subscription_get_topic_name(&(sub->subscription));
+  if (NULL == subscription_name) {
+    Py_RETURN_NONE;
```

Comment:
I think in this case `rcl_subscription_get_topic_name()` woul have set an error message. I think this should raise an exception in that case

example:
https://github.com/ros2/rclpy/blob/78f8a9e3d10964b2c7f347980ebd0e1eb570b526/rclpy/src/rclpy/_rclpy.c#L1973-L1975

Reply:
Whoops, looks like the `RCLError` code got put in the wrong place :upside_down_face: It should be here after the `rcl_subscription_get_topic_name()` call.

Reply:
Oops.  I believe it should be good now.